ipn/ipnlocal/node_backend.go

```diff
@@ -694,8 +694,9 @@ func dnsConfigForNetmap(nm *netmap.NetworkMap, peers map[tailcfg.NodeID]tailcfg.
 	}
 
 	dcfg := &dns.Config{
-		Routes: map[dnsname.FQDN][]*dnstype.Resolver{},
-		Hosts:  map[dnsname.FQDN][]netip.Addr{},
+		Routes:         map[dnsname.FQDN][]*dnstype.Resolver{},
+		Hosts:          map[dnsname.FQDN][]netip.Addr{},
+		SubdomainHosts: map[dnsname.FQDN]bool{},
```

Comment:
Since this will normally be empty, maybe it should start as `nil` and we use `mak.Set` below to allocate the map only when needed?